mitreattack/navlayers/generators/usage_generator.py

```diff
@@ -0,0 +1,128 @@
+        :param raw_matches: matching attack-pattern objects
+        :return: list of dictionary objects for every technique: score=0 if not in raw_matches, 1 otherwise,
+                    description in comments
+        """
+        shortlist = []
+        for match in raw_matches:
+            xid = ''
+            xphase = ''
+            for ref in match.external_references:
+                if ref.source_name == 'mitre-attack':
+                    xid = ref.external_id
+            for phase in match.kill_chain_phases:
+                if phase.kill_chain_name == 'mitre-attack':
+                    xphase = phase.phase_name
+            shortlist.append((xid, xphase, match.description))
+        full_matrix_listing = self.matrix_handle.get_matrix(self.domain)
+        construct = list()
+        for tactic in full_matrix_listing:
+            for tech in tactic.techniques:
+                construct.append(dict(techniqueID=tech.id, score=0,
+                                      tactic=self.matrix_handle.convert(tactic.tactic.name)))
+            for tech_key in tactic.subtechniques:
+                for subtech in tactic.subtechniques[tech_key]:
+                    construct.append(dict(techniqueID=subtech.id, score=0,
+                                          tactic=self.matrix_handle.convert(tactic.tactic.name)))
+        for entry in shortlist:
+            for tac in construct:
+                if entry[0] == tac['techniqueID'] and (entry[1] == '' or entry[1] == tac['tactic']):
+                    tac['score'] = 1
+                    tac['comment'] = entry[2]
+        return construct
+
+    def generate_layer(self, match):
+        """
+        Generate a layer
+        :param match: the pattern to match
+        :return: layer object with annotated techniques
+        """
+        typeChecker(type(self).__name__, match, str, "match")
+        raw_data, matched_obj = self.get_matrix_data(match)
+        if matched_obj.type not in ["course-of-action", 'tool', 'malware', 'intrusion-set']:
+            print(f"Warning: The input match {match} corresponds with an ATT&CK Technique, which is not supported. "
+                  f"Please provide a group, software, or mitigation instead.")
+            raise StixObjectIsNotValid
+        a_id = get_attack_id(matched_obj)
+        processed_listing = self.generate_technique_data(raw_data)
+        raw_layer = dict(name=f"{matched_obj.name} ({matched_obj.id})", domain=self.domain + '-attack')
+        raw_layer['techniques'] = processed_listing
+        output_layer = Layer(raw_layer)
+        if self.domain == 'enterprise':
+            output_layer.description = f"Enterprise techniques used by {matched_obj.name}, " \
+                                       f"ATT&CK {matched_obj.type} {a_id}"
+        else:
+            f"Mobile techniques used by {matched_obj.name}, ATT&CK {matched_obj.type} {a_id}"
```

Comment:
Hardcoding of domain here excludes ICS domain layers.

Reply:
Outcome is probably ICS layers description saying they're a Mobile layer.

Reply:
Ah, that's easy enough to patch. I think this is a holdover from when I did the original development, which didn't consider ICS at the time.